RMET-3875 - Fix issue in permission request for `EditURIPicture` and remove unnecessary permissions

alexgerardojacinto opened the pull request:
## Description
<!--- Describe your changes in detail -->

- Fixes an issue in the permission request in the `callEditUriImage` function, where in some cases the permission kept being requested.
- Removes `READ_MEDIA_IMAGE` and `READ_MEDIA_VIDEO` permissions from the AndroidManifest.xml file of the resulting app, since these are never requested by the plugin, so there's no point in having them in the manifest file.

## Context
<!--- Why is this change required? What problem does it solve? -->
<!--- Place the link to the issue here -->

References: https://outsystemsrd.atlassian.net/browse/RMET-3875

## Type of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply -->
- [x] Fix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Refactor (cosmetic changes)
- [ ] Breaking change (change that would cause existing functionality to not work as expected)

## Platforms affected
- [x] Android
- [ ] iOS
- [ ] JavaScript

## Tests
<!--- Describe how you tested your changes in detail -->
<!--- Include details of your test environment if relevant -->

Tested on Android 15, 14, 13, 12, 11, 10 and 9.

MABS 11 build: https://intranet.outsystems.net/MABS_Lite/BuildDetail?id=7b48b872999002187f70787e92a74b9822616dc3&stg=dev

MABS 10 build: https://intranet.outsystems.net/MABS_Lite/BuildDetail?id=bc959d4b80007a634e3c3d728f2d7b936096e27e&stg=dev

## Screenshots (if appropriate)

## Checklist
<!--- Go over all the following items and put an `x` in all the boxes that apply -->
- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [x] Code follows code style of this project
- [x] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

alexgerardojacinto committed:
fix: properly check if gallery permissions are needed

References: https://outsystemsrd.atlassian.net/browse/RMET-3875

alexgerardojacinto committed:
Merge branch 'development' into fix/RMET-3875/edit-uri-permissions

alexgerardojacinto committed:
fix: avoid requesting unnecessary permission when choosing media from the gallery

Context: When using intents or the Photo Picker API for the user to select media (images and videos) we don't need to request the READ_EXTERNAL_STORAGE permission. That is only necessary when accessing files in the external storage (e.g. when calling editURIPicture)

References: https://outsystemsrd.atlassian.net/browse/RMET-3875

alexgerardojacinto committed:
fix: remove unnecessary permissions from plugin

Context: We actually never request these permissions, so there's no point in declaring them in the AndroidManifest.xml file.

References: https://outsystemsrd.atlassian.net/browse/RMET-3875

alexgerardojacinto committed:
chore: update dependency to OSCameraLib-Android

References: https://outsystemsrd.atlassian.net/browse/RMET-3875

alexgerardojacinto committed:
chore: update changelog

References: https://outsystemsrd.atlassian.net/browse/RMET-3875

OS-pedrogustavobilro commented on src/android/CameraLauncher.kt:
```diff
@@ -469,20 +469,8 @@ class CameraLauncher : CordovaPlugin() {
-            callChooseFromGallery()
-        }
+        
+        callChooseFromGallery()
```
nice 🚀

OS-pedrogustavobilro commented on src/android/CameraLauncher.kt:
```diff
@@ -383,9 +383,9 @@ class CameraLauncher : CordovaPlugin() {
-        val galleryPermissionNeeded = !(Build.VERSION.SDK_INT < 33 &&
-                PermissionHelper.hasPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE) &&
-                PermissionHelper.hasPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE))
+        val galleryPermissionNeeded = Build.VERSION.SDK_INT < 33 &&
+                (!PermissionHelper.hasPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE) ||
+                        (editParameters.saveToGallery && !PermissionHelper.hasPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)))
```
It's not clear to me why the external storage permissions are needed here. I understand that if we save to gallery we save outside the app's storage, but I'm not entirely sure you need storage Permissions - from the code on OSCameraLib-Android, [MediaStore](https://developer.android.com/reference/android/provider/MediaStore) is used - and the docs don't mention requiring permissions for it.

alexgerardojacinto replied:
That's a good point, I believe you're right. After refactoring and running some tests, it seems that these permissions aren't necessary here, so I removed them.

alexgerardojacinto committed:
refactor: avoid asking unnecessary permissions

Context: As we use the MediaStore to fetch and store photos in the device's gallery, these permissions aren't necessary.

References: https://outsystemsrd.atlassian.net/browse/RMET-3875

alexgerardojacinto committed:
chore: set lib to dev version

alexgerardojacinto committed:
Revert "refactor: avoid asking unnecessary permissions"

This reverts commit 28d61e43d10c8c16ab52084dd55fac922fb8cf89.

alexgerardojacinto committed:
revert: revert commit where we removed the READ_EXTERNAL_STORAGE permission request for Android < 13

References: https://outsystemsrd.atlassian.net/browse/RMET-3875

alexgerardojacinto committed:
chore: remove entry from changelog

References: https://outsystemsrd.atlassian.net/browse/RMET-3875

alexgerardojacinto committed:
chore: update dependency to OSCameraLib-Android

References: https://outsystemsrd.atlassian.net/browse/RMET-3875

alexgerardojacinto committed:
chore(release): raise to version 4.2.0-OS52

References: https://outsystemsrd.atlassian.net/browse/RMET-3875
